teps/0062-catalog-tags-and-hub-categories-management.md

```diff
@@ -0,0 +1,195 @@
+
+```yaml
+---
+categories:
+  - name: Build Tools
+    tags: [build-tool]
+  - name: CLI
+    tags: [cli]
+  - name: Cloud
+    tags: [gcp, aws, azure, cloud]
+  - name: Deploy
+    tags: [deploy]
+  - name: Others        👈 Any new tags are mapped here
+    tags: []
+  - name: Test Framework
+    tags: [test]
+```
+
+## Motivation
+
+### Goals
+
+- Make the manual work automated and also properly do the mapping of the resources.
+
+### Non-Goals
+
+### Use Cases (optional)
+
+- Filter the resources on Hub using categories
+- Search the resources based on tags as keywords
+
+## Requirements
+
+- Add category field as annotation in a task
+
+For e.g.
+
+```yaml
+tekton.dev/category: "foo”     👈 Category of a resource
+```
+
+## Proposal
+
+When new tags are added for the resource in the catalog, how these tags should be mapped to categories in Hub. Right now we have to do this manually before every release. As we are doing the catalog refresh at an interval of 30 mins, if this can be automated, will give a better user experience.
+
+For e.g: Consider the following task. It has new tags which are not part of categories of [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml)
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: pylint
+  labels:
+    app.kubernetes.io/version: '0.2'
+  annotations:
+    tekton.dev/pipelines.minVersion: '0.12.1'
+    tekton.dev/tags: python, pylint   👈 New tags
+    tekton.dev/displayName: pylint
+```
+
+### Proposed Solution (Based on Operator Hub)
+
+- Add categories in catalog task as annotation. There should be only one category for a task. Multiple categories are not allowed.
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: pylint
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/category: "foo”     👈 Category of a resource
+    tekton.dev/tags: python, pylint   👈 New tags
+    tekton.dev/displayName: pylint
+```
+
+- CI/Catlin should make sure that category is added for the task and there is only one category. Maintaining the category is the responsibility of the author.
+- If category is not added,Catlin should throw an warning to add a category
+- The category should be added from the predefined category list
+
+  - Sample Predefined Category List:
+
+  ```yaml
+  categories:
+    - Automation
+    - Build Tools
+    - CLI
+    - Cloud
+    - Deploy
+    - Editor
+    - Git
+    - Image Build
+    - Language
+    - Messaging
+    - Monitoring
+    - Notification
+    - Others
+    - Security
+    - Storage
+    - Test Framework
+  ```
+
+- If users wants to add new categories to the above list, the user should create a pull request to [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml)
+
+- Tags can be used in hub as keywords and resources can be searched using tags and can be filtered only by categories
```

Comment:
what's the difference between `searched` and `filtered` as used here?

does searching by tags result in filtering of resources?

Reply:
So currently there's a mapping of categories and tags done by Hub in [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml)
For e.g.
```
---
categories:
  - name: Build Tools
    tags: [build-tool]
  - name: CLI
    tags: [cli]
```

Hence in Hub tasks/pipelines are `filtered` through categories and tasks/pipelines are `searched` through `name/displayName` but according to this proposal categories and tags will be independent of each other and tasks/pipelines will be `filtered` only through `categories` and `searched` through `name`, `displayName` and `tags` as keywords